README.md

```diff
@@ -79,3 +79,66 @@ The current implementation uses goroutines very liberally which means the query
 ### Plan optimization
 
 The current implementation creates a physical plan directly from the PromQL abstract syntax tree. Plan optimizations not yet implemented and would require having a logical plan as an intermediary step.
+
+
+## Latest benchmarks
+
+Single core benchmarks
+```markdown
+vector_selector/current_engine         	      44	  27756507 ns/op	16146356 B/op	   69121 allocs/op
+vector_selector/new_engine             	      32	  36515120 ns/op	25709769 B/op	   76738 allocs/op
+
+sum/current_engine                     	      22	  47253905 ns/op	 4925751 B/op	   73843 allocs/op
+sum/new_engine                         	      42	  28137694 ns/op	 6933936 B/op	   71399 allocs/op
+
+sum_by_pod/current_engine              	       8	 135361672 ns/op	87259230 B/op	  568616 allocs/op
+sum_by_pod/new_engine                  	      30	  36838789 ns/op	16031030 B/op	  156454 allocs/op
+
+rate/current_engine                    	      24	  49954635 ns/op	17207644 B/op	   78176 allocs/op
+rate/new_engine                        	      20	  57753527 ns/op	28160190 B/op	  103748 allocs/op
```

Comment:
We know why it is worse in this case? `rate`, `vector`, etc

Reply:
I believe we are worse because we iterate through the data twice for this case, while the new engine has to do it once.

The matrix selector operator produces step vectors, and the engine takes those vectors and converts them into points. Since there's a 1 to 1 mapping between the input and output data, we are wasting lots of time in this conversion. 

I don't think this is an issue though, people do not really query high cardinality data with raw selectors.

Reply:
I am not very familiar with that part of code. Is this possible for us to handle the raw selectors specially and scan the data only once?

Reply:
Yes, this should be possible, especially if we have a planner. We can have a different execution path just for expressions that only have a selectors.